### ISSUE TITLE
there should be an extra option to lock numerals to western script

### ISSUE THREAD

**Issue by itt533** [ibus/ibus#2530]
Which distribution and version?: 
slackware 64 15.0

Which desktop environment and version?: 
xfce 4.16.0

Which session type?: 
X11

Which application and version?: 
any in which text can be entered

IBus version?: 
1.5.25

Issue description:
when selecting an indic input method like 'hindi', 'telugu' one automatically
gets numerals too in those languages script. Since nearly all countries, and 
language based communities have adopted the western numerals, there 
should be an independent option allowing or disabling the chosen language 
numerals. That is how it is in latex, and libreoffice.

Steps to reproduce:
1. chose for example 'itrans' or 'inscript telugu' input method
1. type a number like 4, or 5
1. ౪, or ౫ will be displayed

Can you reproduce your problem when you restart ibus-daemon? (yes / no): yes

Do you see any errors when you run ibus-daemon with the verbose option?:
Can you reproduce your problem with a new user account instead of the current your account? (yes / no): 

it's a wish, therefore these 2 questions are not applicable

**Comment by mike-fabian** [comment]
That is not an ibus issue but an issue of the m17n-db input methods involved.

These m17n-db input methods can have options which you can set in the setup tools of ibus-m17n or ibus-typing-booster (both ibus-m17n and ibus-typing-booster can use all m17n-db input methods, I recommend to use ibus-typing-booster).

But to make that possible that input methods needs to have an option.

For input methods with options look at these examples (look for "variable" in the source):

https://git.savannah.nongnu.org/cgit/m17n/m17n-db.git/tree/MIM/mr-gamabhana.mim

which has an option `trim-last-halant` or

https://git.savannah.nongnu.org/cgit/m17n/m17n-db.git/tree/MIM/bn-national-jatiya.mim

which has an option `use-automatic-vowel-forming`.

None of the currently existing input methods for Telugu (te-itrans.mim, te-inscript.mim, te-inscript2.mim ...) has any options.

It would be possible to add such an option (which might be called `use-western-numbers` or something like that.

To discuss that (how that option should be named, which input methods would need it, ...), please open an issue at: 

https://savannah.nongnu.org/bugs/?group=m17n

and/or post to the mailing list:

https://lists.nongnu.org/mailman/listinfo/m17n-list

**Comment by fujiwarat** [comment]
Can you report your issue to ibus-m17n instead of ibus?
https://github.com/ibus/ibus/wiki/BugReport

I guess it might be an option to provide a shortcut key to toggle the language numerals and ASCII digits.

**Comment by mike-fabian** [comment]
It would be possible to introduce new options in ibus-m17n and ibus-typing-booster to convert all language specific digits to ASCII digits. I.e. if the input method used produces a language specific digit, it would be converted to the ASCII digit if that option is on. 

Something like:

☑️ Convert language specific digits to ASCII

I am not sure whether that is a good idea though. 

The advantage of that idea would be that it would work for *all* m17n-db input methods without having to change each input method. Another advantage would be that in ibus-typing-booster I could quite easily implement this as an option which could be toggled with a user definable key binding (Not in ibus-m17n though, ibus-m17n does not have user definable key bindings). So this would be useful if one wants to switch often between languages specific digits and ASCII digits.

It would also be rather easy to implement in ibus-typing-booster, but much more difficult in ibus-m17n, so I would probably implement that *only* in ibus-typing-booster.

The other idea I suggested in my last comment https://github.com/ibus/ibus/issues/2530#issuecomment-1598663239 needs a change in *every* input method which wants that feature. If this feature is wanted for `te-inscript2`, then `te-inscript2.mim` needs to be changed.

Looking at `te-inscript2.mim` as an example:

https://git.savannah.nongnu.org/cgit/m17n/m17n-db.git/tree/MIM/te-inscript2.mim

It contains for example:

```
  ("4" "౪")
  ((G-4) "₹")
```

`G-4` means `AltGr+4`. So one could not use `G-<something>` to produce the ASCII digit as many `G-<something>` are already in use for something else.

Therefore, I suggested to add an option like `use-western-numbers` to the `te-inscript2.mim` file and if that option is set to `1`, then typing the `4` key would produce the ASCII 4, if it is set to `0` then typing the `4` key would produce the Telugu number `౪`.

Setting this variable in ibus-typing-booster would look as in the example videos in this issue:

https://github.com/mike-fabian/ibus-typing-booster/issues/360

(And it would look very similar in ibus-m17n, ibus-m17n can also change such input method specific variables in the setup tool).

The advantage of this way to implement it would be that changing `te-inscript2.mim` like that would make it work in both ibus-m17n and ibus-typing-booster without any changes in these input engines. 

It would not be possible to toggle this with key binding though, one would need to use the setup tool. So this would be fine if one almost always sticks to using only one type of number and doesn't switch back and forth very often.

I am undecided at the moment which is best, I tend to think it is better to introduce such an option in every m17n-db input method where this is wanted, i.e. change `te-inscript2.mim`, `te-inscript.mim`, `te-itrans.mim`, ...

Actually one could even do both, have an option 

☑️ Convert language specific digits to ASCII

in ibus-typing-booster *and* introduce options in some `.mim` files. 

I just wrote this for brainstorming here, please open a new issue either at 

https://github.com/mike-fabian/ibus-typing-booster/issues

for the possible new ibus-typing-booster option and/or at 

https://savannah.nongnu.org/bugs/?group=m17n

to request changes to the m17n-db input methods.

**Comment by fujiwarat** [comment]
>  ibus-m17n does not have user definable key bindings

OK, then, the shortcut key would be another priority. The panel menu item only might be still useful at the moment for ibus-m17n.
I'm also thinking another option to toggle US layout and the current layout within ibus-m17n.

> te-inscript2.mim needs to be changed

I'm not sure if the feature should depend on m17n-db

**Comment by mike-fabian** [comment]
@itt533 Would you be willing to test a “☑️ Convert language specific digits to ASCII” feature in ibus-typing-booster if I implement that?

**Comment by itt533** [comment]
it's ok i'll test it

**Comment by itt533** [comment]
do i still need to report this wish to https://lists.nongnu.org/mailman/listinfo/m17n-list ?

**Comment by fujiwarat** [comment]
> do i still need to report this wish to https://lists.nongnu.org/mailman/listinfo/m17n-list ?

No. You should report your issue to ibus-m17n instead.
Using space key is a bad option now.

**Comment by mike-fabian** [comment]
I have moved this issue to the ibus-typing-booster issue tracker:

https://github.com/mike-fabian/ibus-typing-booster/issues/445

**Comment by mike-fabian** [comment]
@itt533 Please have a look at https://github.com/mike-fabian/ibus-typing-booster/issues/445

I have added a video showing the new option.

**Comment by mike-fabian** [comment]
@itt533 

I made another update adding a command which can be bound to a key to toggle the new “☑️ Convert language specific digits to ASCII digits” option with a keyboard shortcut

See: https://github.com/mike-fabian/ibus-typing-booster/issues/445#issuecomment-1609180283
